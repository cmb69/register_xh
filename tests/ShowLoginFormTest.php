<?php

/**
 * Copyright (c) 2021-2023 Christoph M. Becker
 *
 * This file is part of Register_XH.
 */

namespace Register;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Register\Infra\Request;
use Register\Infra\Url;
use Register\Infra\UserRepository;
use Register\Infra\View;
use Register\Value\User;

class ShowLoginFormTest extends TestCase
{
    public function testLoginForm()
    {
        $plugin_cf = XH_includeVar("./config/config.php", 'plugin_cf');
        $conf = $plugin_cf['register'];
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $text = $plugin_tx['register'];
        $userRepository = $this->createStub(UserRepository::class);
        $subject = new ShowLoginForm($conf, $text, $userRepository, new View("./views/", $text));

        $request = $this->createStub(Request::class);
        $request->method("url")->willReturn(new Url("/", "Foo"));
        $response = $subject($request);

        Approvals::verifyHtml($response->output());
    }

    public function testLoggedInForm()
    {
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $text = $plugin_tx['register'];
        $user = new User("jane", "", [], "Jane Doe", "jane@example.com", "activated", "secret");
        $userRepository = $this->createStub(UserRepository::class);
        $userRepository->method("findByUsername")->willReturn($user);
        $subject = new ShowLoginForm([], $text, $userRepository, new View("./views/", $text));

        $request = $this->createStub(Request::class);
        $request->method("url")->willReturn(new Url("/", "Foo"));
        $request->method("username")->willReturn("jane");
        $response = $subject($request);

        Approvals::verifyHtml($response->output());
    }
}
