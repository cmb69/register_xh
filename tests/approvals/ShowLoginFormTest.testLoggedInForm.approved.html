<!-- register logged in area -->
<div class="regi_loggedin_loggedinarea">
  <div class="regi_loggedin_user">Hello Jane Doe!</div>
  <div class="regi_loggedin_loggedin">Logged in</div>
  <div class="regi_loggedin_settings">
    <a href="/?User-Preferences" title="User Preferences">
      User Preferences    </a>
  </div>
  <div class="regi_loggedin_logout">
    <a href="/?&amp;function=registerlogout" title="Logout">
      Logout    </a>
  </div>
  <div style="clear: both;"></div>
</div>
